src/main/java/org/tl/Hobbies.java

```diff
@@ -21,8 +35,28 @@ public List<String> createListFromFile(String filePath) {
      * @param filePath The path to the file to read.
      * @return A map where each person's name is associated with a list of their hobbies.
      */
-    public Map<String, List<String>> createDictionary(String filePath) {
-        return null;
+    public Map<String, Set<String>> createDictionary(String filePath) {
+
+        Map<String, Set<String>> result = new HashMap<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(":");
+                if (parts.length != 2) {
+                    throw new IllegalArgumentException("Invalid format in the input file");
+                }
+                String name = parts[0];
+                String[] hobbyArray = parts[1].split(",");
+                Set<String> hobbies = new HashSet<>(Arrays.asList(hobbyArray));
+                result.merge(name, hobbies, (existingHobbies, newHobbies) -> {
+                    existingHobbies.addAll(newHobbies);
+                    return existingHobbies;
+                });
```

Comment:
You don't need to read from file again, use the `createListFromFile` to get the list with lines and then iterate over that list